client/src/components/Profile.jsx

```diff
@@ -4,18 +4,89 @@ import Payment from './Payment.jsx';
 import FeedContainer from './FeedContainer.jsx';
 import ProfileHeader from './ProfileHeader.jsx';
 import axios from 'axios';
+import feedManipulation from '../feedManipulation.js'
 
 class Profile extends React.Component {
   constructor (props) {
     super(props);
     this.state = {
       profileInfo: {},
-      unknownUser: false
+      unknownUser: false,
+      profileFeed: {},
+      relationalFeed: {}
     }
   }
 
   componentDidMount() {
-    this.loadProfileData(this.props.match.params.username);
+    let profileUsername = this.props.match.params.username;
+    let userId = this.props.userInfo.userId;
+    this.loadProfileData(profileUsername);
+    this.getProfileFeeds('profileFeed', userId, null, profileUsername);
+    this.getProfileFeeds('relationalFeed', userId, null, profileUsername);
+  }
+
+  getProfileFeeds(feedType, userId = null, sinceId, profileUsername) {
+    let endpoint = feedManipulation.returnFeedEndpoint(feedType, userId);
+
+    let params = {
+      sinceId: sinceId,
+      userId: userId,
+      profileUsername: profileUsername
+    }
+
+    axios(endpoint, {params: params})
+      .then((response) => {
+        this.prependNewTransactions(feedType, response.data);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  prependNewTransactions(feedType, transactionSummary) {
+    // If no results return, do nothing
+    if (!transactionSummary || transactionSummary.count === 0) {
+      return;
+    }
+
+    // If feed was empty, set the returned transactions as the feed
+    let isFeedEmpty = !this.state[feedType].count || this.state[feedType].count === 0;
+
+    let newFeedObject = isFeedEmpty
+      ? transactionSummary
+      : feedManipulation.mergeFeeds(transactionSummary, this.state[feedType]);
+
+    this.setState({
+      [feedType]: newFeedObject
```

Comment:
Why the square brackets?

Reply:
@acenine - it causes Javascript to evaluate the variable feedType, rather than setting a key at the string feedType.